lib/manifest-parser.js

```diff
@@ -24,16 +25,33 @@ function getVersionFromVersionString(nameAndVersion) {
 }
 exports.getVersionFromVersionString = getVersionFromVersionString;
 
+function isYarnLockFileV2(manifest) {
+  return manifest.match(/__metadata:/)
+}
+
 exports.parseDependencies = function(manifest) {
-  var dependencies = yarn.parse(manifest).object
+  if (isYarnLockFileV2(manifest)) {
+    var dependencies = yaml.load(manifest, {
+      onWarning: function(err) { console.log('YAML parsing error: ', err) },
+      json: true, // JSON compat mode will overwrite dupe entries instead of throw an error.
+      schema: yaml.FAILSAFE_SCHEMA,
+    });
+  } else {
+    var dependencies = yarn.parse(manifest).object
+  }
+
   var deps = []
   Object.keys(dependencies).forEach((dep) => {
+    if (dep === '__metadata') { return; }
     deps.push({
       name: getNameFromVersionString(dep),
       version: dependencies[dep].version,
-      requirement: getVersionFromVersionString(dep),
+      requirement: getVersionFromVersionString(dep), // TODO: parse  tweetnacl double requirement example properly here, and use comparator intersection set (https://classic.yarnpkg.com/lang/en/docs/dependency-versions/#toc-intersections)
       type: 'runtime'
     })
   })
+
   return deps;
 }
+
+
```

Comment:
```suggestion
  })
  return deps;
}
```

some extra line cleanup from editor! :)